test/rules/no-consecutive-blank-lines/test.ts.lint

```diff
@@ -14,20 +14,20 @@ class Clazz { // comment
 
 ~nil
         console.log("test");
-~nil                         [consecutive blank lines are disallowed]
+~nil [consecutive blank lines are disallowed]
 
     }
 
 
 ~nil
 }
-~nil  [consecutive blank lines are disallowed]
+~nil [consecutive blank lines are disallowed]
 
 //Begin whitespace
 // The next two lines of "code" contain only tabs or spaces, they are also considered "blank" lines
-~nil                         [consecutive blank lines are disallowed]
         
         
+~ [consecutive blank lines are disallowed]
```

Comment:
Oh believe me, we used to do things that way, and we had comments from people (such as TS team members) who said they wanted to contribute to TSLint but that writing tests was too onerous.

The downsides of that approach are pretty painful: you have to manually figure out the position and length of each lint failure. Then, if you change your file at all, you have to go through and update the positions for every expected failure.

It's true that the current system doesn't work so well for this rule. But for other rules [it works nicely](https://github.com/palantir/tslint/blob/master/test/rules/triple-equals/allow-null-check/test.ts.lint).

Our testing system is inspired by the way the TS team does tests. I'd recommend [reading this issue](https://github.com/Microsoft/TypeScript/issues/6528) for some great pointers from them on testing, it's an interesting read